### ISSUE TITLE
Compile fails in MSVC with /clr

### ISSUE THREAD

**Issue by MrSapps** [FlorianRappl/CmdParser#17]
Seems you can't have a namespace called "cli" when building with /clr.

`error C3292: the cli namespace cannot be reopened`

**Comment by FlorianRappl** [comment]
Yeah the Microsoft compiler (esp. with the CLR target) was never the main objective / target for this library, but supporting the build would be desirable. I just fear it would be a breaking change. Maybe have a macro depending on the compiler / flags to switch the namespace (i.e., use "cli" for non-CLR target and "ncli" or similar for CLR)?

Not sure if there is a compiler directive check we can apply here (which one). Would be happy if somebody could come up with a solution / PR. Any help appreciated!

**Comment by MrSapps** [comment]
Yeah I'm just sure.. I just hacked my local copy by renaming the namespace..

**Comment by zoulunkai** [comment]
Despite successfully building the project without /clr, I encountered debugging issues. Specifically, the MSVC2022 debugger couldn't inspect variables within the Parser class in debug mode. Renaming the "cli" namespace resolved this issue.

**Comment by FlorianRappl** [comment]
Again as per my comment if there is a directive that only applies in case of MSVC we could circumvent this. Otherwise this is more a bug in MSVC as it's not behaving standards compliant.
